### ISSUE TITLE
2.1.1 Configurable product "%1" does not have sub-products

### ISSUE THREAD

**Issue by paales** [magento/magento2#6422]
### Preconditions

1. I've upgraded to a Magento 2.1.1 installation
### Steps to reproduce

1. Create a configurable where the product with the lowest price isn't in stock
2. Hide products in stock
3. Now search the product on the frontend.
### Expected result

1. It doesn't use the product that isn't in stock / selects all products or something
### Actual result

<details> 
  <summary>

(Click me): #0 (Magento\Framework\Exception\LocalizedException): Configureerbaar product "100174" heeft geen sub-producten</summary>

```
1 exception(s):
Exception #0 (Magento\Framework\Exception\LocalizedException): Configureerbaar product "100174" heeft geen sub-producten

Exception #0 (Magento\Framework\Exception\LocalizedException): Configureerbaar product "100174" heeft geen sub-producten
#0 /vendor/magento/module-configurable-product/Pricing/Price/ConfigurableRegularPrice.php(71): Magento\ConfigurableProduct\Pricing\Price\ConfigurablePriceResolver->resolvePrice(Object(Ho\Promopost\Model\Product\Interceptor))
#1 /vendor/magento/framework/Pricing/Price/AbstractPrice.php(98): Magento\ConfigurableProduct\Pricing\Price\ConfigurableRegularPrice->getValue()
#2 /vendor/magento/module-configurable-product/Pricing/Price/ConfigurableRegularPrice.php(121): Magento\Framework\Pricing\Price\AbstractPrice->getAmount()
#3 /vendor/magento/module-configurable-product/Pricing/Price/ConfigurableRegularPrice.php(82): Magento\ConfigurableProduct\Pricing\Price\ConfigurableRegularPrice->getMinRegularAmount()
#4 /vendor/magento/module-catalog/Pricing/Render/FinalPriceBox.php(101): Magento\ConfigurableProduct\Pricing\Price\ConfigurableRegularPrice->getAmount()
#5 /var/generation/Magento/Catalog/Pricing/Render/FinalPriceBox/Interceptor.php(37): Magento\Catalog\Pricing\Render\FinalPriceBox->hasSpecialPrice()
#6 /app/code/Ho/TierPrices/view/frontend/templates/catalog/product/final_price.phtml(26): Magento\Catalog\Pricing\Render\FinalPriceBox\Interceptor->hasSpecialPrice()
#7 /vendor/magento/framework/View/TemplateEngine/Php.php(59): include('/Users/paul/Sit...')
#8 /vendor/magento/framework/View/Element/Template.php(255): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Catalog\Pricing\Render\FinalPriceBox\Interceptor), '/Users/paul/Sit...', Array)
#9 /var/generation/Magento/Catalog/Pricing/Render/FinalPriceBox/Interceptor.php(258): Magento\Framework\View\Element\Template->fetchView('/Users/paul/Sit...')
#10 /vendor/magento/framework/View/Element/Template.php(279): Magento\Catalog\Pricing\Render\FinalPriceBox\Interceptor->fetchView('/Users/paul/Sit...')
#11 /vendor/magento/framework/Pricing/Render/PriceBox.php(69): Magento\Framework\View\Element\Template->_toHtml()
#12 /vendor/magento/module-catalog/Pricing/Render/FinalPriceBox.php(31): Magento\Framework\Pricing\Render\PriceBox->_toHtml()
#13 /vendor/magento/framework/View/Element/AbstractBlock.php(659): Magento\Catalog\Pricing\Render\FinalPriceBox->_toHtml()
#14 /var/generation/Magento/Catalog/Pricing/Render/FinalPriceBox/Interceptor.php(570): Magento\Framework\View\Element\AbstractBlock->toHtml()
#15 /vendor/magento/framework/Pricing/Render.php(99): Magento\Catalog\Pricing\Render\FinalPriceBox\Interceptor->toHtml()
#16 /vendor/magento/module-catalog/Block/Product/ListProduct.php(364): Magento\Framework\Pricing\Render->render('final_price', Object(Ho\Promopost\Model\Product\Interceptor), Array)
#17 /app/design/frontend/Ho/promopost/Magento_Catalog/templates/product/list.phtml(88): Magento\Catalog\Block\Product\ListProduct->getProductPrice(Object(Ho\Promopost\Model\Product\Interceptor))
#18 /vendor/magento/framework/View/TemplateEngine/Php.php(59): include('/Users/paul/Sit...')
#19 /vendor/magento/framework/View/Element/Template.php(255): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Catalog\Block\Product\ListProduct), '/Users/paul/Sit...', Array)
#20 /vendor/magento/framework/View/Element/Template.php(279): Magento\Framework\View\Element\Template->fetchView('/Users/paul/Sit...')
#21 /vendor/magento/framework/View/Element/AbstractBlock.php(659): Magento\Framework\View\Element\Template->_toHtml()
#22 /vendor/magento/framework/View/Layout.php(542): Magento\Framework\View\Element\AbstractBlock->toHtml()
#23 /vendor/magento/framework/View/Layout.php(518): Magento\Framework\View\Layout->_renderBlock('search_result_l...')
#24 /var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('search_result_l...')
#25 /vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('search_result_l...')
#26 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Layout->renderElement('search_result_l...', true)
#27 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Layout\Interceptor->___callParent('renderElement', Array)
#28 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderElement', Object(Magento\Framework\View\Layout\Interceptor), Array, 'hoTemplatehints...')
#29 /vendor/ho-nl/magento2-templatehints/Plugin/View/LayoutPlugin.php(77): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}('search_result_l...', true)
#30 /vendor/magento/framework/Interception/Interceptor.php(142): Ho\Templatehints\Plugin\View\LayoutPlugin->aroundRenderElement(Object(Magento\Framework\View\Layout\Interceptor), Object(Closure), 'search_result_l...', true)
#31 /var/generation/Magento/Framework/View/Layout/Interceptor.php(195): Magento\Framework\View\Layout\Interceptor->___callPlugins('renderElement', Array, Array)
#32 /vendor/magento/framework/View/Element/AbstractBlock.php(499): Magento\Framework\View\Layout\Interceptor->renderElement('search_result_l...', true)
#33 /vendor/magento/module-catalog-search/Block/Result.php(168): Magento\Framework\View\Element\AbstractBlock->getChildHtml('search_result_l...')
#34 /vendor/magento/module-catalog-search/view/frontend/templates/result.phtml(22): Magento\CatalogSearch\Block\Result->getProductListHtml()
#35 /vendor/magento/framework/View/TemplateEngine/Php.php(59): include('/Users/paul/Sit...')
#36 /vendor/magento/framework/View/Element/Template.php(255): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\CatalogSearch\Block\Result), '/Users/paul/Sit...', Array)
#37 /vendor/magento/framework/View/Element/Template.php(279): Magento\Framework\View\Element\Template->fetchView('/Users/paul/Sit...')
#38 /vendor/magento/framework/View/Element/AbstractBlock.php(659): Magento\Framework\View\Element\Template->_toHtml()
#39 /vendor/magento/framework/View/Layout.php(542): Magento\Framework\View\Element\AbstractBlock->toHtml()
#40 /vendor/magento/framework/View/Layout.php(518): Magento\Framework\View\Layout->_renderBlock('search.result')
#41 /var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('search.result')
#42 /vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('search.result')
#43 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Layout->renderElement('search.result', false)
#44 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Layout\Interceptor->___callParent('renderElement', Array)
#45 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderElement', Object(Magento\Framework\View\Layout\Interceptor), Array, 'hoTemplatehints...')
#46 /vendor/ho-nl/magento2-templatehints/Plugin/View/LayoutPlugin.php(77): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}('search.result', false)
#47 /vendor/magento/framework/Interception/Interceptor.php(142): Ho\Templatehints\Plugin\View\LayoutPlugin->aroundRenderElement(Object(Magento\Framework\View\Layout\Interceptor), Object(Closure), 'search.result')
#48 /var/generation/Magento/Framework/View/Layout/Interceptor.php(195): Magento\Framework\View\Layout\Interceptor->___callPlugins('renderElement', Array, Array)
#49 /vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('search.result')
#50 /vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('content')
#51 /var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('content')
#52 /vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('content')
#53 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Layout->renderElement('content', false)
#54 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Layout\Interceptor->___callParent('renderElement', Array)
#55 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderElement', Object(Magento\Framework\View\Layout\Interceptor), Array, 'hoTemplatehints...')
#56 /vendor/ho-nl/magento2-templatehints/Plugin/View/LayoutPlugin.php(77): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}('content', false)
#57 /vendor/magento/framework/Interception/Interceptor.php(142): Ho\Templatehints\Plugin\View\LayoutPlugin->aroundRenderElement(Object(Magento\Framework\View\Layout\Interceptor), Object(Closure), 'content')
#58 /var/generation/Magento/Framework/View/Layout/Interceptor.php(195): Magento\Framework\View\Layout\Interceptor->___callPlugins('renderElement', Array, Array)
#59 /vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('content')
#60 /vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('main')
#61 /var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('main')
#62 /vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('main')
#63 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Layout->renderElement('main', false)
#64 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Layout\Interceptor->___callParent('renderElement', Array)
#65 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderElement', Object(Magento\Framework\View\Layout\Interceptor), Array, 'hoTemplatehints...')
#66 /vendor/ho-nl/magento2-templatehints/Plugin/View/LayoutPlugin.php(77): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}('main', false)
#67 /vendor/magento/framework/Interception/Interceptor.php(142): Ho\Templatehints\Plugin\View\LayoutPlugin->aroundRenderElement(Object(Magento\Framework\View\Layout\Interceptor), Object(Closure), 'main')
#68 /var/generation/Magento/Framework/View/Layout/Interceptor.php(195): Magento\Framework\View\Layout\Interceptor->___callPlugins('renderElement', Array, Array)
#69 /vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('main')
#70 /vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('columns')
#71 /var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('columns')
#72 /vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('columns')
#73 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Layout->renderElement('columns', false)
#74 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Layout\Interceptor->___callParent('renderElement', Array)
#75 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderElement', Object(Magento\Framework\View\Layout\Interceptor), Array, 'hoTemplatehints...')
#76 /vendor/ho-nl/magento2-templatehints/Plugin/View/LayoutPlugin.php(77): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}('columns', false)
#77 /vendor/magento/framework/Interception/Interceptor.php(142): Ho\Templatehints\Plugin\View\LayoutPlugin->aroundRenderElement(Object(Magento\Framework\View\Layout\Interceptor), Object(Closure), 'columns')
#78 /var/generation/Magento/Framework/View/Layout/Interceptor.php(195): Magento\Framework\View\Layout\Interceptor->___callPlugins('renderElement', Array, Array)
#79 /vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('columns')
#80 /vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('main.content')
#81 /var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('main.content')
#82 /vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('main.content')
#83 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Layout->renderElement('main.content', false)
#84 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Layout\Interceptor->___callParent('renderElement', Array)
#85 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderElement', Object(Magento\Framework\View\Layout\Interceptor), Array, 'hoTemplatehints...')
#86 /vendor/ho-nl/magento2-templatehints/Plugin/View/LayoutPlugin.php(77): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}('main.content', false)
#87 /vendor/magento/framework/Interception/Interceptor.php(142): Ho\Templatehints\Plugin\View\LayoutPlugin->aroundRenderElement(Object(Magento\Framework\View\Layout\Interceptor), Object(Closure), 'main.content')
#88 /var/generation/Magento/Framework/View/Layout/Interceptor.php(195): Magento\Framework\View\Layout\Interceptor->___callPlugins('renderElement', Array, Array)
#89 /vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('main.content')
#90 /vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('page.wrapper')
#91 /var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('page.wrapper')
#92 /vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('page.wrapper')
#93 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Layout->renderElement('page.wrapper', false)
#94 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Layout\Interceptor->___callParent('renderElement', Array)
#95 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderElement', Object(Magento\Framework\View\Layout\Interceptor), Array, 'hoTemplatehints...')
#96 /vendor/ho-nl/magento2-templatehints/Plugin/View/LayoutPlugin.php(77): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}('page.wrapper', false)
#97 /vendor/magento/framework/Interception/Interceptor.php(142): Ho\Templatehints\Plugin\View\LayoutPlugin->aroundRenderElement(Object(Magento\Framework\View\Layout\Interceptor), Object(Closure), 'page.wrapper')
#98 /var/generation/Magento/Framework/View/Layout/Interceptor.php(195): Magento\Framework\View\Layout\Interceptor->___callPlugins('renderElement', Array, Array)
#99 /vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('page.wrapper')
#100 /vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('root')
#101 /var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('root')
#102 /vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('root')
#103 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Layout->renderElement('root', false)
#104 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Layout\Interceptor->___callParent('renderElement', Array)
#105 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderElement', Object(Magento\Framework\View\Layout\Interceptor), Array, 'hoTemplatehints...')
#106 /vendor/ho-nl/magento2-templatehints/Plugin/View/LayoutPlugin.php(77): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}('root', false)
#107 /vendor/magento/framework/Interception/Interceptor.php(142): Ho\Templatehints\Plugin\View\LayoutPlugin->aroundRenderElement(Object(Magento\Framework\View\Layout\Interceptor), Object(Closure), 'root')
#108 /var/generation/Magento/Framework/View/Layout/Interceptor.php(195): Magento\Framework\View\Layout\Interceptor->___callPlugins('renderElement', Array, Array)
#109 /vendor/magento/framework/View/Layout.php(938): Magento\Framework\View\Layout\Interceptor->renderElement('root')
#110 /vendor/magento/framework/Interception/Interceptor.php(146): Magento\Framework\View\Layout->getOutput()
#111 /var/generation/Magento/Framework/View/Layout/Interceptor.php(494): Magento\Framework\View\Layout\Interceptor->___callPlugins('getOutput', Array, Array)
#112 /vendor/magento/framework/View/Result/Page.php(243): Magento\Framework\View\Layout\Interceptor->getOutput()
#113 /vendor/magento/framework/View/Result/Layout.php(164): Magento\Framework\View\Result\Page->render(Object(Magento\Framework\App\Response\Http\Interceptor))
#114 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Result\Layout->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#115 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Result\Page\Interceptor->___callParent('renderResult', Array)
#116 /vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderResult', Object(Magento\Framework\View\Result\Page\Interceptor), Array, 'result-varnish-...')
#117 /vendor/magento/module-page-cache/Model/Controller/Result/VarnishPlugin.php(74): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Response\Http\Interceptor))
#118 /vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\PageCache\Model\Controller\Result\VarnishPlugin->aroundRenderResult(Object(Magento\Framework\View\Result\Page\Interceptor), Object(Closure), Object(Magento\Framework\App\Response\Http\Interceptor))
#119 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderResult', Object(Magento\Framework\View\Result\Page\Interceptor), Array, 'result-builtin-...')
#120 /vendor/magento/module-page-cache/Model/Controller/Result/BuiltinPlugin.php(67): Magento\Framework\View\Result\Page\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Response\Http\Interceptor))
#121 /vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\Controller\Result\BuiltinPlugin->aroundRenderResult(Object(Magento\Framework\View\Result\Page\Interceptor), Object(Closure), Object(Magento\Framework\App\Response\Http\Interceptor))
#122 /var/generation/Magento/Framework/View/Result/Page/Interceptor.php(130): Magento\Framework\View\Result\Page\Interceptor->___callPlugins('renderResult', Array, Array)
#123 /vendor/magento/framework/App/View.php(221): Magento\Framework\View\Result\Page\Interceptor->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#124 /vendor/magento/module-catalog-search/Controller/Result/Index.php(91): Magento\Framework\App\View->renderLayout()
#125 /var/generation/Magento/CatalogSearch/Controller/Result/Index/Interceptor.php(24): Magento\CatalogSearch\Controller\Result\Index->execute()
#126 /vendor/magento/framework/App/Action/Action.php(102): Magento\CatalogSearch\Controller\Result\Index\Interceptor->execute()
#127 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#128 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\CatalogSearch\Controller\Result\Index\Interceptor->___callParent('dispatch', Array)
#129 /vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'dispatch', Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Array, 'contextPlugin')
#130 /vendor/magento/module-store/App/Action/Plugin/Context.php(106): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#131 /vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\Context->aroundDispatch(Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#132 /vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'dispatch', Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Array, 'customer-app-ac...')
#133 /vendor/magento/module-customer/Model/App/Action/ContextPlugin.php(61): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#134 /vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Customer\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#135 /vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'dispatch', Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Array, 'storeCheck')
#136 /vendor/magento/module-store/App/Action/Plugin/StoreCheck.php(44): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#137 /vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\StoreCheck->aroundDispatch(Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#138 /vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'dispatch', Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Array, 'weee-app-action...')
#139 /vendor/magento/module-weee/Model/App/Action/ContextPlugin.php(112): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#140 /vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Weee\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#141 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'dispatch', Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Array, 'tax-app-action-...')
#142 /vendor/magento/module-tax/Model/App/Action/ContextPlugin.php(91): Magento\CatalogSearch\Controller\Result\Index\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#143 /vendor/magento/framework/Interception/Interceptor.php(142): Magento\Tax\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\CatalogSearch\Controller\Result\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#144 /var/generation/Magento/CatalogSearch/Controller/Result/Index/Interceptor.php(39): Magento\CatalogSearch\Controller\Result\Index\Interceptor->___callPlugins('dispatch', Array, Array)
#145 /vendor/magento/framework/App/FrontController.php(55): Magento\CatalogSearch\Controller\Result\Index\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#146 /vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#147 /vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#148 /vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'requestPreproce...')
#149 /vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php(94): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#150 /vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#151 /vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#152 /vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#153 /vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#154 /vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#155 /vendor/magento/module-page-cache/Model/App/FrontController/VarnishPlugin.php(55): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#156 /vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\PageCache\Model\App\FrontController\VarnishPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#157 /vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#158 /vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(68): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#159 /vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#160 /var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#161 /vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#162 /vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#163 /index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#164 {main}
```

</details>

https://github.com/magento/magento2/blob/308da7d18a46d2a3e4dc8a553e38f5ec58b476f4/app/code/Magento/ConfigurableProduct/Pricing/Price/ConfigurableOptionsProvider.php#L75

The problem is that the UNION select has a limit on it, causing not all products to be selected..

**Comment by MorganGonzales** [comment]
This happens to me as well after I upgraded from 2.1 to 2.1.1.

**Comment by heldchen** [comment]
I've just reported a similar issue but with "state" instead of "stock": #6447

**Comment by sevos1984** [comment]
Created internal ticket MAGETWO-58033. Thanks for reporting

**Comment by cuiyang000** [comment]
Can confirm the bug in 2.1.1

**Comment by nmallepally** [comment]
I have the same error 
Hi,
I have the same issue after upgrading from 2.1.0 to 2.1.1 via composer.
configurable products does not show anymore on frontend.
when i checked the exception log, this is what it shows me 
[2016-09-21 08:23:25] main.CRITICAL: Magento\Framework\Exception\LocalizedException: Configurable product "kit_pro" does not have sub-products in /var/www/vhosts/magento2.1.1/httpdocs/test123/vendor/magento/module-configurable-product/Pricing/Price/ConfigurablePriceResolver.php:65
Stack trace:
#0 /var/www/vhosts/magento2.1.1/httpdocs/test123/vendor/magento/module-configurable-product/Pricing/Price/FinalPrice.php(43): Magento\ConfigurableProduct\Pricing\Price\ConfigurablePriceResolver->resolvePrice(Object(Magento\Catalog\Model\Product\Interceptor))

Attached a file
front end errors.txt
of errors displayed on frontend saying it does not have sub-products, but on the backend it has products associated with it. I tried clearing browser cache and reindexing it, did not resolve the issue.
[front.end.errors.txt](https://github.com/magento/magento2/files/484664/front.end.errors.txt)

**Comment by immist** [comment]
@sevos1984 ist there a timeline for fixing MAGETWO-58922 (#6799/#5762) and MAGETWO-56480 (#5948)?
Because of these CRITICAL bugs at the moment Magento 2 is useless for any standard ecommerce site using configurable products.

**Comment by ewallkrishnaprakash** [comment]
I also face the same error.

Hi ,

I have upgrade my magento from 2.0.7 to 2.1.1.
After that i face error in the following scenario.
-> I was add configurable product with 5 child products (combination of colors and sizes).
-> After that if any one of the child product is goes to out of stock.
-> When open respective category (Ie., the configurable product associated with which category) in front-end it show blank in content section. the layered navigation,header,footer are load perfectly.
-> While check in log files i get an error **"[2016-10-13 09:36:02] main.CRITICAL: Configurable product "SS019-007-014" does not have sub-products [] []"**
-> I was google it by this issue, in magento forums some one says if we disable or make instock to the respective child product then it will be solved.
-> I thought it is not a proper solution, so some one help me, how to resolve this issue in a right way.

**Comment by JamboMedia** [comment]
Any update on this? We upgraded to magento 2.1.2 and we have the same problem.

The products are not shown anymore at the category pages. Very important that this will be fixed soon.

**Comment by markdavies** [comment]
This is a critical error - any update?

**Comment by jakobfdev** [comment]
Please fix this soon.
My project is becoming an odyssey of reading up on bugs...

**Comment by bharatgunani** [comment]
Please fix this soon.

**Comment by magnetic5355** [comment]
This bug also causes product pages to return the error whenever all sub products are out of stock causing visitors from Google etc. to receive errors instead of out of stock notice.

**Comment by TommyKolkman** [comment]
Confirming this bug in 2.1.2 as well. Is there a quick fix available? We've got a project that is going live real soon and I can't wait for this... Pretty critical.

**Comment by paales** [comment]
For a workaround, see https://github.com/magento/magento2/issues/5762#issuecomment-253480901

**Comment by ThomasNegeli** [comment]
Seems to work in 2.1.3 now. Can you confirm?

**Comment by maksek** [comment]
Fixed in 2.1.3, and in 2.2. Feel free to ping me if it still reproducible.
